sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/ReplicatedResourceClient.java

```diff
@@ -158,48 +155,94 @@ public Mono<StoreResponse> invokeAsync(RxDocumentServiceRequest request,
 
     private Mono<StoreResponse> getStoreResponseMonoWithSpeculation(RxDocumentServiceRequest request, Quadruple<Boolean, Boolean, Duration, Integer> forceRefreshAndTimeout) {
         CosmosEndToEndOperationLatencyPolicyConfig config = request.requestContext.getEndToEndOperationLatencyPolicyConfig();
+        List<String> preferredRegions = this.transportClient.getGlobalEndpointManager().getPreferredRegions();
+        AvailabilityStrategy strategy = config.getAvailabilityStrategy();
+        if (strategy == null){
+            strategy = new DefaultAvailabilityStrategy();
+        }
+        if (strategy instanceof ThresholdBasedAvailabilityStrategy) {
+                speculativeProcessor = new ThresholdBasedSpeculation();
+        }
+
         List<Mono<StoreResponse>> monoList = new ArrayList<>();
         List<RxDocumentServiceRequest> requests = new ArrayList<>();
 
-        if (speculativeProcessor.shouldIncludeOriginalRequestRegion()) {
+        if (speculativeProcessor != null) {
+            List<String> effectiveRetryRegions = strategy.getEffectiveRetryRegions(preferredRegions,
+                request.requestContext.getExcludeRegions());
+            if (effectiveRetryRegions.size() > BridgeInternal.getNumberOfRegionsToTry(strategy)){
+                effectiveRetryRegions = effectiveRetryRegions.subList(0, BridgeInternal.getNumberOfRegionsToTry(strategy));
+            }
+            effectiveRetryRegions
+                .forEach(regionName -> {
+                    URI locationURI = getLocationURIFromAvailableEndPoints(regionName, request);
```

Comment:
I thought the excludeRegion list is independent of the endToEndTimeoutPolicy? That even if customer does not config the e2eTimeoutPolicy, but the excludeRegion list will still be honored by other retry policies? (like ClientRetryPolicy)

Reply:
Discussed offline, honoring in other retry policy will happen in another following up PR